File: code/design-patterns/python/structural/bridge/bridge.py
from abc import ABC, abstractmethod


class Printer(ABC):
    @abstractmethod
    def print(self):
        pass


class HPPrinter(Printer):
    def print(self):
        print("Printing by an HP printer")


class EpsonPrinter(Printer):
    def print(self):
        print("Printing by an Epson printer")


class Computer(ABC):
    def __init__(self, printer: Printer):
        self.printer = printer

    @abstractmethod
    def print(self):
        pass


class Mac(Computer):
    def print(self):
        print("Mac computer printing using", end=" ")
        self.printer.print()


class Windows(Computer):
    def print(self):
        print("Windows computer printing using", end=" ")
        self.printer.print()

File: code/design-patterns/structural/bridge/go/bridge.go
package bridge

import "fmt"

// Abstraction Layer

type Computer struct {
	printer Printer
}

func (c *Computer) SetPrinter(printer Printer) {
	c.printer = printer
}

func (c *Computer) Print() {
	c.printer.Print()
}

type Mac struct {
	Computer
}

type Windows struct {
	Computer
}

// Implementor Layer

type Printer interface {
	Print()
}

type HPPrinter struct{}

func (p *HPPrinter) Print() {
	fmt.Println("Printing by an HP printer")
}

type EpsonPrinter struct{}

func (p *EpsonPrinter) Print() {
	fmt.Println("Printing by an Epson printer")
}
